apps/epic-react/src/components/lesson-completion-toggle.tsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react'
+import {motion} from 'framer-motion'
+import {useReward} from 'react-rewards'
+import {useLesson} from '@skillrecordings/skill-lesson/hooks/use-lesson'
+import {useRouter} from 'next/router'
+import {trpcSkillLessons} from '@skillrecordings/skill-lesson/utils/trpc-skill-lessons'
+import toast from 'react-hot-toast'
+
+const LessonCompleteToggle = () => {
+  const {module} = useLesson()
+  const router = useRouter()
+  const {reward, isAnimating} = useReward('rewardId', 'confetti')
+  const [toggleClicked, setToggleClicked] = React.useState(false)
+  const lessonSlug = router.query.lesson
+  const {
+    data: moduleProgress,
+    status: moduleProgressStatus,
+    isFetching,
+  } = trpcSkillLessons.moduleProgress.bySlug.useQuery({
+    slug: module.slug.current,
+  })
+  const moduleCompleted = moduleProgress?.moduleCompleted
+
+  const toggleProgressMutation = trpcSkillLessons.progress.toggle.useMutation()
+  const completedLessons = moduleProgress?.lessons.filter(
+    (l: any) => l.lessonCompleted,
+  )
+
+  // cannot use id because lesson from useLesson returns solution resource when on solution page
+  const isLessonCompleted = Boolean(
+    completedLessons?.find(({id, slug}: any) => slug === lessonSlug),
+  )
+
+  const [optimisticallyToggled, setOptimisticallyToggled] = React.useState(
+    isLessonCompleted || false,
+  )
+
+  React.useEffect(() => {
+    moduleProgressStatus === 'success' &&
+      setOptimisticallyToggled(isLessonCompleted)
+  }, [moduleProgressStatus, isLessonCompleted])
+
+  React.useEffect(() => {
+    if (toggleClicked && moduleCompleted) {
+      router.push(`/modules/${module.slug.current}/completed`)
+    } else if (toggleClicked && optimisticallyToggled) {
+      reward()
+      router.push(
+        `/modules/${module.slug.current}/${moduleProgress?.nextLesson?.slug}`,
+      )
+    }
+  }, [toggleClicked, moduleProgress, optimisticallyToggled])
```

Comment:
watching if the completion button is clicked, if it is we check if the module is completed. If the module is completed we navigate to the `[module]/completed` route. If it isn't then we navigate to the next lesson.